### PR TITLE
add associations

### PR DESCRIPTION
**Pull request by EyasuTeshome**
In this PR:
- Created a class classroom.
- Created the has-many/belongs-to relationship between Classroom and Student.
- Created a class Book.
- Created a class Rental.
- Created the many-to-many relationship between Person and Book using the intermediate class Rental.

### CODE REVIEW

**Review by Ismailco — COMMENTED** [review]
Hi @EyasuTeshome, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has no professional documentation, kindly consider using the [README template](https://github.com/microverseinc/readme-template/blob/master/README.md) provided by microverse to Create your README file

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: school/book.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Book
+  attr_accessor :title, :author
```
Comment (tufoinnkuo10) [inline school/book.rb]: - [ ] kindly add @rentals as it is the class used to set the relationship between person and book

File: school/rental.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class Classroom
```
Comment (tufoinnkuo10) [inline school/rental.rb]: - [ ] Great job. kindly add a rental class
- [ ] kindly add its setters and getters for the @date instance variable, and ensure that the @date instance variable is initialized in the constructor

File: school/person.rb [hunk 3]
```diff
@@ -24,4 +25,8 @@ def correct_name
   def of_age?
     @age >= 18
   end
+
```
Comment (tufoinnkuo10) [inline school/person.rb:30]: - [ ] it is rental of a book. The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association. something like this
```ruby
def rent_book(book, date)
    Rental.new(date, book, self)
  end
```

File: school/student.rb [hunk 4]
```diff
@@ -7,4 +7,9 @@ def initialize(age, _name, _parent_permission, classroom = nil)
   def play_hooky
     "¯\(ツ)/¯"
   end
+
```
Comment (tufoinnkuo10) [inline school/student.rb:14]: - [ ] also add classroom to class student
`attr_reader :classroom`

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @EyasuTeshome 

There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Github flow
✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally

Required Changes ♻️
Check the comments under the review
Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Review by KDlamini — APPROVED** [review]
### Hi @EyasuTeshome 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## HIGHLIGHTS 🌟 

✔️ All required classes present
✔️ Inheritance and associations implemented
✔️ Good PR description
✔️ No linter errors

# Status: Approved ✔️ 

simon

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**